### ISSUE TITLE
 Implement a glossary system for novel-specific word replacement

### ISSUE THREAD

**Issue by bregassa** [LNReader/lnreader#965]
### Describe your suggested feature

MTL is cancer, but we can make it better by having a glossary system on the apps.
I would like to request the addition of a glossary system to LNReader. This feature would allow users to create and manage their own glossaries for each novel they read. This would be especially useful for reading machine-translated (MTL) novels, which often contain incorrect or unfamiliar terminology.

Here's how the glossary system could work:

- On each novel page, users would be able to add new words and phrases to their glossary, along with their corresponding correct or preferred terminologies.
- The glossary would be applied automatically while reading a novel, replacing glossary terms with their corresponding preferred word value the user has written.
- Users would be able to enable/disable the glossary for each novel individually.

Benefits:

- This feature would improve the reading experience for users who read from MTL novel sources.
- It would also be useful for users who are learning a new language, as they could create glossaries of new words they encounter.
- Additionally, it would allow users to personalize their reading experience by creating custom glossaries for specific novels if for example the terminology used in comic and novel is different and the reader prefers to use comic terminology, so the reader can just add his glossary entries.

Screenshot example of the similar tool from Firefox add-on called FoxReplace : 
![image](https://github.com/LNReader/lnreader/assets/22976574/78a1d63e-5ec6-40da-84d0-89d286bd4764)

### Other details

Might be useful
https://github.com/Woundorf/foxreplace

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with a source, I should be opening an issue in the [sources repository](https://github.com/LNReader/lnreader-sources/issues/new/choose).
- [X] I have updated the app to version **[1.1.19](https://github.com/LNReader/lnreader/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

**Comment by Pheonix-Flames** [comment]
Yes it would be very good as I have to update my javascript constantly
